libyaml_q_declaration.md

```diff
@@ -0,0 +1,135 @@
+
+Considering the previously mentioned reasons, we consider this library to be robust and reliable. In terms of ROS2 package metrics to be Quality Level 4.
+
+Even though `libyaml` by itself will not likely reach the equivalent level of quality as Quality Level 1, there are steps that can be taken by ROS contributors to ensure that its incorporation into ROS packages can provide the equivalent level of quality.
+
+ROS contributors will need to conduct coverage tests to identify the remaining API and features that are not currently covered by tests. If features and portions of the public API that are used in ROS dependencies are found to be untested, then the appropriate tests will be required. Including specific tests for the `libyaml` version imported by `libyaml_vendor` for the tier 1 platforms listed in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers).
+
+# Comparison with ROS packages quality standards
+
+## Version policy [1]
+
+### Version Scheme [1.i]
+
+It is not stated if the library supports any kind of version policy.
+
+### Version Stability [1.ii]
+    
+Current version of `libyaml` in [its repository](https://github.com/yaml/libyaml) is  0.24, however, for the purposes of ROS2 Quality Level analysis, the imported version of `libyaml` through the `libyaml_vendor` package is fixed to the 0.18 version.
+
+### Public API Declaration [1.iii]
+    
+As a C library, elements available in its `yaml.h` are considered to be their public API.
+
+### API Stability Policy [1.iv]
+    
+There is no policy for API stability. This is not a problem because the `libyaml_vendor` package importing the `libyaml dependency` is using a fixed version, in this case, the [0.18](https://github.com/yaml/libyaml/tree/release-0.1.8)
+
+### ABI Stability Policy [1.v]
+    
+There is no policy for ABI stability. This is not a problem because the `libyaml_vendor` package importing the `libyaml` dependency is using a fixed version, in this case, the [0.18](https://github.com/yaml/libyaml/tree/release-0.1.8)
+
+### ABI and ABI Stability Within a Released ROS Distribution [1.vi]
+    
+There is not a direct correlation between the `libyaml` releases and the ROS distributions, however this is not a problem because the `libyaml_vendor` package importing the `libyaml` dependency is using a fixed version, in this case, the [0.18](https://github.com/yaml/libyaml/tree/release-0.1.8)
+
+
+## Change Control Process [2]
+
+### Change Requests [2.i]
+    
+Checking through the commits history, it can be seen is not the case.
+
+### Contributor Origin [2.ii]
+    
+Does not have it (or it does not seem like it’s the case).
+
+### Peer Review Policy [2.iii]
+    
+Seems to be followed for pull requests on the Github repository, but as not all code changes occur through change requests, this can not be confirmed for these changes.
+
+### Continuous Integration [2.iv]
+
+If any, it not publicly available.
```

Comment:
some comment here that we are building it in the ROS2 buildfarm. what do you think?

Reply:
I think CI builds the latest upstream version, which is not the same as the fixed one imported in `libyaml_vendor`. Eventually, tor reach a higher quality level, there should be some testing for this package itself though.